### PR TITLE
feature: logic of login by email and by Google

### PR DESCRIPTION
**Pull request by ivanderkachov**
Provide logic of login by email and by Google
Using Redux RTK Query, @react-oauth/google library

Снимок экрана 2022-11-16 в 15 22 52
Снимок экрана 2022-11-16 в 15 23 57
Снимок экрана 2022-11-16 в 15 24 24

### CODE REVIEW

File: apps/freelance/src/redux/auth.api.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  id: string;
+  email: string;
+  isActivated: boolean;
+}
+
+interface IToken {
+  token: string;
+}
+
+const baseUrl: string = process.env['NX_API_URL_AUTH'] || '';
```
Comment (lyudad) [inline apps/freelance/src/redux/auth.api.ts]: it should be in constant file. in other case this condition will be in each API file, it does not make any sence

File: libs/components/src/lib/form-container/form-container.tsx [hunk 2]
```diff
@@ -34,6 +36,15 @@ export function FormContainer({
   forgotPassLink,
 }: FormContainerProps) {
   const { t } = useTranslation();
+  const [loginWithGoogle] = useLoginWithGoogleMutation();
+
+  const loginGoogle = useGoogleLogin({
+    onSuccess: async coderesponse => {
+      const response = await loginWithGoogle({ token: coderesponse.access_token });
```
Comment (lyudad) [inline libs/components/src/lib/form-container/form-container.tsx]: Снимок экрана 2022-11-16 в 22 37 07

File: libs/components/src/lib/form-container/form-container.tsx [hunk 3]
```diff
@@ -60,7 +71,15 @@ export function FormContainer({
...
       {isSignForm && (
-        <StyledButton id="googleBtn" buttonSize="md" buttonColor="blue">
+        <StyledButton
+          id="googleBtn"
+          buttonSize="md"
+          buttonColor="blue"
+          onClick={() => {
+            
+            loginGoogle();
+          }}
```
Comment (lyudad) [inline libs/components/src/lib/form-container/form-container.tsx]: onClick={loginGoogle}

File: apps/freelance/src/redux/auth.api.ts (hunk 1 above)
Comment (ivanderkachov) [reply apps/freelance/src/redux/auth.api.ts]: Updated

File: libs/components/src/lib/form-container/form-container.tsx (hunk 2 above)
Comment (ivanderkachov) [reply libs/components/src/lib/form-container/form-container.tsx]: Updated

File: libs/components/src/lib/form-container/form-container.tsx (hunk 3 above)
Comment (ivanderkachov) [reply libs/components/src/lib/form-container/form-container.tsx]: Updated

File: libs/components/src/lib/login-form/login-form.tsx [hunk 4]
```diff
@@ -21,16 +23,26 @@ export function LoginForm(props: LoginFormProps) {
     resolver: yupResolver(signInSchema),
   });
 
-  const formSubmitHandler = (data:Props) => {
-    console.log(data);
-    //use rtk query in future of course
+  const formSubmitHandler = (data: Props) => {
+
```
Comment (lyudad) [inline libs/components/src/lib/login-form/login-form.tsx:27]: why this space needed?

**Pull request closed without merging**